File: tonghua-project/backend/app/models/campaign.py
from sqlalchemy import Column, Integer, String, DateTime, Text, Enum, DECIMAL, func
from app.database import Base


class Campaign(Base):
    __tablename__ = "campaigns"

    id = Column(Integer, primary_key=True, autoincrement=True)
    title = Column(String(300), nullable=False)
    description = Column(Text, nullable=True)
    cover_image = Column(String(500), nullable=True)
    start_date = Column(DateTime, nullable=False)
    end_date = Column(DateTime, nullable=False)
    goal_amount = Column(DECIMAL(12, 2), nullable=False)
    current_amount = Column(DECIMAL(12, 2), default=0, nullable=False)
    status = Column(
        Enum("draft", "active", "completed", "cancelled", name="campaign_status"),
        default="draft",
        nullable=False,
    )
    participant_count = Column(Integer, default=0, nullable=False)
    artwork_count = Column(Integer, default=0, nullable=False)
    created_at = Column(DateTime, server_default=func.now(), nullable=False)

File: frontend/android/app/src/main/java/org/tonghua/app/data/model/Campaign.kt
package org.tonghua.app.data.model

import com.google.gson.annotations.SerializedName

/**
 * Campaign model for charity art campaigns.
 */
data class Campaign(
    @SerializedName("id") val id: String,
    @SerializedName("title") val title: String,
    @SerializedName("theme") val theme: String,
    @SerializedName("description") val description: String,
    @SerializedName("cover_image_url") val coverImageUrl: String?,
    @SerializedName("start_date") val startDate: String,
    @SerializedName("end_date") val endDate: String,
    @SerializedName("status") val status: String,
    @SerializedName("max_artworks") val maxArtworks: Int?,
    @SerializedName("artwork_count") val artworkCount: Int,
    @SerializedName("vote_enabled") val voteEnabled: Boolean,
)
